tools/ds-identify

```diff
@@ -179,10 +179,30 @@ debug() {
     echo "$@" 1>&3
 }
 
+map_sysfs_to_dmi() {
+    case "$@" in
+        sys_vendor) _RET="system-manufacturer";;
+        product_name) _RET="system-product-name";;
+        product_uuid) _RET="system-uuid";;
+        product_serial) _RET="system-serial-number";;
+        chassis-asset-tag) _RET="chassis_asset_tag";;
+        *) return 1;;
+    esac
+}
+
 get_dmi_field() {
     local path="${PATH_SYS_CLASS_DMI_ID}/$1"
     if [ ! -f "$path" ] || [ ! -r "$path" ]; then
-        _RET="$UNAVAILABLE"
+        map_sysfs_to_dmi "$1" || {
+            _RET="$ERROR: unknown field $1"
+            return
+        }
+        local dmi_field="$_RET" dmi_value=""
+        dmi_value=$(dmidecode --quiet --string "$dmi_field") || {
```

Comment:
perhaps we should throw away `stderr` here, since it's a lot:

<details>
<summary>dmidecode --quiet --string blah-blubb</summary>

```
Invalid string keyword: blah-blubb
Valid string keywords are:
  bios-vendor
  bios-version
  bios-release-date
  system-manufacturer
  system-product-name
  system-version
  system-serial-number
  system-uuid
  baseboard-manufacturer
  baseboard-product-name
  baseboard-version
  baseboard-serial-number
  baseboard-asset-tag
  chassis-manufacturer
  chassis-type
  chassis-version
  chassis-serial-number
  chassis-asset-tag
  processor-family
  processor-manufacturer
  processor-version
  processor-frequency
```

</details>

Reply:
IIUC, we should only see this if we've misconfigured `ds-identify` to map to an invalid field. So I think it's fine (and perhaps desirable) to have more verbose output in an error case.

(If I don't understand correctly, please do correct me!)

Reply:
ooor, if our shell code passes thru some values from untrustworthy sources

Reply:
> ooor, if our shell code passes thru some values from untrustworthy sources

```
$ a --no-heading get_dmi_field
tools/ds-identify:182:get_dmi_field() {
tools/ds-identify:306:    get_dmi_field chassis_asset_tag
tools/ds-identify:312:    get_dmi_field sys_vendor
tools/ds-identify:318:    get_dmi_field product_name
tools/ds-identify:324:    get_dmi_field product_uuid
tools/ds-identify:330:    get_dmi_field product_serial
```

Looks to me like we're fine from that POV.